paperqa/agents/search.py

```diff
@@ -258,9 +258,12 @@ async def index_files(self) -> dict[str, str]:
             if await file_index_path.exists():
                 async with await anyio.open_file(file_index_path, "rb") as f:
                     content = await f.read()
-                    self._index_files = pickle.loads(  # noqa: S301
-                        zlib.decompress(content)
-                    )
+                    try:
+                        self._index_files = pickle.loads(  # noqa: S301
+                            zlib.decompress(content)
+                        )
+                    except Exception:
```

Comment:
We need to re-`raise`, as suppressing the `Exception` (what your PR does) changes the behavior

Reply:
Isn't it 'nicer' to just ignore the corrupt index file than fail?

Reply:
These `incomplete or truncated stream` correspond with empty or missing files, which means degraded performance of the index. The degradation's impact depends on how valuable the missing contents were.

Let's say the missing file contains the relevant information to answer, so you get a false negative: PaperQA would get the answer incorrect, but just because the relevant information was gone from the index.

So it's fine to add this logging of the impacted file, but we don't want to allow degraded performance in this part of the stack.

Reply:
Okay, added the reraising now.